src/phongo_structs.h

```diff
@@ -177,6 +177,29 @@ typedef struct {
 	zend_object std;
 } php_phongo_binary_t;
 
+typedef struct {
+	bson_t*     bson;
+	HashTable*  properties;
+	zend_object std;
+} php_phongo_bsonarray_t;
+
+typedef struct {
+	bson_t*     bson;
+	HashTable*  properties;
+	zend_object std;
+} php_phongo_bsondocument_t;
+
+typedef struct {
+	zval        bson;
```

Comment:
I'll have to revisit this: we may get away with only storing a copy of the `bson_t`, and even if we need the entire zval a `zval*` coupled with properly placed reference counting should suffice.

Reply:
After double-checking the implementation, there's nothing left for us to do. We don't accidentally free anything too early, and we also don't unnecessarily copy BSON structures around.